ga4gh/backend.py

```diff
@@ -229,8 +245,13 @@ def _getDatasetFromVariantsRequest(self, request):
         return dataset
 
     def _getDatasetFromCompoundId(self, compoundId):
+        if compoundId is None:
+            raise exceptions.BadIdentifierException(compoundId)
+        if not hasattr(compoundId, 'split'):
```

Comment:
this is just testing compoundId is a string, basically, but I agree it could be better phrased